tests/ci/merge_pr.py

```diff
@@ -32,59 +33,60 @@ def __init__(self, pr: PullRequest):
         """
         logging.info("Checking the PR for approvals")
         self.pr = pr
-        self.reviews = pr.get_reviews()
-        # the reviews are ordered by time
-        self._review_per_user = {}  # type: Dict[NamedUser, PullRequestReview]
-        self.approved_at = datetime.fromtimestamp(0)
-        for r in self.reviews:
+        reviews = pr.get_reviews()
+        # self.reviews is a dict of latest CHANGES_REQUESTED or APPROVED review
+        # per user
+        # NamedUsed has proper __eq__ and __hash__, so it's safe to use it
+        self.reviews = {}  # type: Dict[NamedUser, PullRequestReview]
+        for r in reviews:
             user = r.user
             if r.state not in self.STATES:
                 continue
 
-            if r.state == "APPROVED":
-                self.approved_at = max(r.submitted_at, self.approved_at)
-
-            if not self._review_per_user.get(user):
-                self._review_per_user[user] = r
+            if not self.reviews.get(user):
+                self.reviews[user] = r
                 continue
 
-            if r.submitted_at < self._review_per_user[user].submitted_at:
-                self._review_per_user[user] = r
+            if r.submitted_at < self.reviews[user].submitted_at:
+                self.reviews[user] = r
 
     def is_approved(self, team: List[NamedUser]) -> bool:
         """Checks if the PR is approved, and no changes made after the last approval"""
         if not self.reviews:
             logging.info("There aren't reviews for PR #%s", self.pr.number)
             return False
 
+        filtered_reviews = {
+            user: review
+            for user, review in self.reviews.items()
+            if review.state in self.STATES and user in team
```

Comment:
we already filter by self.STATES in the constructor, I think we can remove this check from here. what do you think?

Reply:
I'd rather remove the condition from line 43, since here it's called a `filtered_reviews`, and we can leave all types of reviews in `self.reviews`